src/library/overviewcache.cpp

```diff
@@ -0,0 +1,156 @@
+#include "library/overviewcache.h"
+
+#include <QFutureWatcher>
+#include <QPixmapCache>
+#include <QSqlDatabase>
+#include <QtConcurrentRun>
+#include <QtSql>
+
+#include "library/dao/analysisdao.h"
+#include "moc_overviewcache.cpp"
+#include "util/db/dbconnectionpooled.h"
+#include "util/db/dbconnectionpooler.h"
+#include "util/logger.h"
+#include "waveform/overviews/waveformoverviewrenderer.h"
+#include "waveform/waveformfactory.h"
+
+namespace {
+
+mixxx::Logger kLogger("OverviewCache");
+
+QString pixmapCacheKey(TrackId trackId, QSize size, WOverview::Type type) {
+    return QString("Overview_%1_%2_%3_%4")
+            .arg(static_cast<int>(type))
+            .arg(trackId.toString())
+            .arg(size.width())
+            .arg(size.height());
+}
```

Comment:
using a string as a cache key is not particularly efficient. a struct with a `qHash` implementation would be better.

Reply:
(I think most of this is adopted from CoverArtCache)

Reply:
What about this?
https://doc.qt.io/qt-6/qpixmapcache-key.html#details
> Use [QPixmapCache::insert](https://doc.qt.io/qt-6/qpixmapcache.html#insert)() to receive an instance of Key generated by the pixmap cache. You can store the key in your own objects for a very efficient one-to-one object-to-pixmap mapping.

Does such a one-to-one map QString <--> Key map/hash make sense performance wise?

Reply:
Probably not, you're right. I didn't see that this was using QPixmapCache internally. The only improvement then is 
```suggestion
QString pixmapCacheKey(TrackId trackId, QSize size, WOverview::Type type) {
    return QStringLiteral("Overview_%1_%2_%3_%4")
            .arg(static_cast<int>(type),
                trackId.toString(),
                size.width(),
                size.height());
}
```